components/fxa-client/android/src/main/java/mozilla/appservices/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fxa
+
+import mozilla.appservices.fxaclient.AccessTokenInfo
+import mozilla.appservices.fxaclient.Device
+import mozilla.appservices.fxaclient.DeviceCapability
+import mozilla.appservices.fxaclient.DevicePushSubscription
+import mozilla.appservices.fxaclient.FxaConfig
+import mozilla.appservices.fxaclient.FirefoxAccount as RustFirefoxAccount
+import mozilla.appservices.fxaclient.IncomingDeviceCommand
+import mozilla.appservices.fxaclient.Profile
+import mozilla.appservices.sync15.DeviceType
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Top-level class to manage a FxA account
+ *
+ * This class contains the top-level functionality like fetching tokens and listening for account
+ * events.  See also:
+ *   - StateManager: manages the account state machine.  Defined in `State.kt` and available from
+ *     the `stateManager()` method.
+ *   - DeviceManager: manages the list of connected devices to an FxA account.  Defined in
+ *     `Devices.kt` and available from the `deviceManager()` method.
+ */
+abstract class FirefoxAccount(
```

Comment:
I don't quite understand why we'd want an abstract class like this in our repo and why these private vals exist? eg, is `applicationScopes` how we'd obtain the list of scopes we care about and how does it relate to the `singleScope` params used below?

Reply:
It's abstract just because I didn't want to write the code.  I don't think it should be in the final version.  And the `private` keyword was a mistake, I think all of those should be constructor arguments.